### PR TITLE
Feature/rdm 12992 delete multi linked cases

### PR DESCRIPTION
**Pull request by sem-hmcts**
### JIRA link (if applicable) ###
[RDM-12856](https://tools.hmcts.net/jira/browse/RDM-12856) _"Functional Tests for the Case Deletion Process"_
[RDM-12992](https://tools.hmcts.net/jira/browse/RDM-12992) _"Delete multiple linked cases in the ccd-case-disposer"_
[RDM-13046](https://tools.hmcts.net/jira/browse/RDM-13046) _"Linked cases deletion causing an unexpected error"_

### Change description ###
Refactor logic to identify cases due deletion and delete them.
Also implement functional tests

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by mattnayler — CHANGES_REQUESTED** [review]
Only minor review comments.

File: src/functionalTest/java/uk/gov/hmcts/reform/ccd/config/TestApplicationConfiguration.java [hunk 1]
```diff
@@ -24,14 +25,19 @@
...
 public class TestApplicationConfiguration {
 
-    @Value("#{'${test.case.types}'.split(',')}")
-    private List<String> testCaseTypes;
+    @Bean
+    public DataSource provideDataSource() {
+        return DataSourceBuilder.create()
+            .url("jdbc:postgresql://localhost:5050/ccd_data?stringtype=unspecified")
+            .username("ccd")
+            .password("ccd")
```
[the remark this answers is not in the document]
Comment (sem-hmcts) [reply src/functionalTest/java/uk/gov/hmcts/reform/ccd/config/TestApplicationConfiguration.java:33]: Resolved by commit efe65d45f515fec388a8874be3d2e1d34e0db209 under RDM-13045

File: src/functionalTest/java/uk/gov/hmcts/reform/ccd/TestDataProvider.java [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+                return null;
+            }
+            final Map<String, Object> sourceAsMap = getResponse.getSourceAsMap();
+            return (Long) sourceAsMap.get(CASE_REFERENCE_FIELD);
+        }));
+    }
+
+    private String getIndex(String caseType) {
+        return String.format("%s_cases", caseType.toLowerCase());
+    }
```
[the remark this answers is not in the document]
Comment (sem-hmcts) [reply src/functionalTest/java/uk/gov/hmcts/reform/ccd/TestDataProvider.java:365]: No, I don't think we want to do that.  I think we want to, as much as possible, test the production code as a black box.  This is here to verify that the production code is doing what we want it to do.

File: src/functionalTest/resources/scenarios/S-001.sql [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        'CaseCreated',
+        'PUBLIC',
+        '{
+          "PersonFirstName": "Janet"
+         }',
+        '{
+          "PersonFirstName": "PUBLIC"
+         }',
+        1504259907353529,
+        '2016-06-24 20:44:52.824'
```
[the remark this answers is not in the document]
Comment (sem-hmcts) [reply src/functionalTest/resources/scenarios/S-001.sql:18]: We could but if we want to do it this way then I suggest we do it as part of RDM-13046

**Review by mattnayler — APPROVED** [review]
LGTM including merger of RDM-13046

**Pull request merged**